src/full/Agda/TypeChecking/Conversion.hs

```diff
@@ -1693,14 +1693,22 @@ equalSort s1 s2 = do
     let (s1,s2) = (ignoreBlocking s1b, ignoreBlocking s2b)
         blocker = unblockOnEither (getBlocker s1b) (getBlocker s2b)
 
-    let postponeIfBlocked = catchPatternErr $ \blocker ->
-          if | blocker == neverUnblock -> typeError $ UnequalSorts s1 s2
+    let postponeIfBlocked = catchPatternErr $ \ blocker -> do
+
+          if | blocker == neverUnblock  -> typeError $ UnequalSorts s1 s2
+-- --             | blocker == alwaysUnblock -> equalSort (ignoreBlocking s1b) (ignoreBlocking s2b)
              | otherwise -> do
                  reportSDoc "tc.conv.sort" 30 $ vcat
                    [ "Postponing constraint"
                    , nest 2 $ fsep [ prettyTCM s1 <+> "=="
                                    , prettyTCM s2 ]
                    ]
+
+                 -- Andreas, 2023-12-21, recomputing the blocker fixes issue #7034.
+                 s1b <- reduceB s1
+                 s2b <- reduceB s2
+                 let blocker = unblockOnEither (getBlocker s1b) (getBlocker s2b)
+
```

Comment:
Is there a good reason to use `reduce` here rather than `updateBlocker`? That would avoid doing the reduction twice (once here, and once the constraint gets woken up again). Alternatively, you could use the reduced sorts in the call to `addConstraint` below.

Reply:
No, I didn't known about `updateBlocker`...

Reply:
It doesn't exactly do the same as what you are doing here. Instead, if the meta in the `Blocker` has been instantiated, it will simply replace it with `alwaysUnblock`. That way the constraint will be retried at the earliest opportunity (which will then reduce the sorts, possibly producing a new blocker).